### ISSUE TITLE
CRAN check returns WARNING in ogr_core.h

### ISSUE THREAD

**Issue by caiohamamura** [rwinlib/gdal2#9]
I've submitted a package using the 2.2.3 release version, but CRAN threw a WARNING and refused my package. The ogr_core.h when checked with -Wpedantic will throw the error:

ogr_core.h:394:19: warning: ISO C restricts enumerator values to range of 'int' [-Wpedantic]

Is there another GDAL version (major 2) which will not trigger this warning?

![image](https://user-images.githubusercontent.com/3440255/76761151-a7a62080-676d-11ea-82da-981b02527a1b.png)

**Comment by jeroen** [comment]
Hmm which package is this? Why don't we see this for any other pkgs? Can you give me the link to the full check output?

**Comment by caiohamamura** [comment]
https://win-builder.r-project.org/incoming_pretest/rGEDI_0.0.1_20200315_044000/Windows/00check.log

**Comment by jeroen** [comment]
This looks like a bug in gdal that still appears in the latest versions of gdal 2 and 3. Can you report this in https://github.com/OSGeo/gdal ?

By the way the same compiler warning appears on Debian, but somehow this does not generate a WARNING: https://win-builder.r-project.org/incoming_pretest/rGEDI_0.0.1_20200315_044000/Debian/00install.out

**Comment by caiohamamura** [comment]
Yes, I realized that Debian wasn't complaining, I thought it was because it was using a different version of gdal, I didn't look into the install.out, strange...

I'll report an issue in gdal, though.

Thanks!

**Comment by jeroen** [comment]
Yeah it's pretty difficult to solve otherwise. Is there any way you can avoid including the `ogr_srs_api.h` header api in your `gedisimulator/gediIO.o` or is this a critical header?

**Comment by caiohamamura** [comment]
Well, actually this is already coming from Steven Hancock's library, it's no my own, but I could tweak his code. I will do it and include a #ifdef WITH_GDAL or something
